### ISSUE TITLE
Support abstract structured logger interface

### ISSUE THREAD

**Issue by zchee** [go-language-server/protocol#23]
Currently, the logger depends `go.uber.org/zap` package.

Should provide the abstract (but structured) `Logger` interface for developers who use the protocol package.

Like [go-logr/logr](https://github.com/go-logr/logr), but Language Server Protocol specific.

**Comment by stefanvanburen** [comment]
Should this package just target [slog](https://pkg.go.dev/log/slog) as the logger?

**Comment by zchee** [comment]
@stefanvanburen 
Yes, When I create this issue, not exist slog. But slog has been one of Go community de-fact logger.
I planned switch to using slog instead of go-logr.

Current status of this repository, See also:
https://github.com/go-language-server/protocol/issues/48#issuecomment-2427327653

**Comment by emcfarlane** [comment]
Fixed by https://github.com/go-language-server/protocol/pull/59
